### PR TITLE
Added Tiberium Fiend hiding

### PR DESCRIPTION
**Pull request by Mailaender**
Chose a rather simple way of setting it up.

### CODE REVIEW

File: mods/ts/rules/critters.yaml [hunk 1]
```diff
@@ -36,6 +36,17 @@ DOGGIE:
 	HitShape:
 		Type: Circle
 			Radius: 213
+	GrantConditionOnTerrain:
+		Condition: hidingplace
+		TerrainTypes: Tiberium, BlueTiberium
+	GrantConditionOnMovement:
+		Condition: moving
+	WithIdleOverlay:
```
Comment (reaperrr) [inline mods/ts/rules/critters.yaml]: `WithSpriteBody` should work as well and would be a bit more fitting.

Comment (Mailaender) [reply mods/ts/rules/critters.yaml]: You are right.

**Comment by reaperrr** [comment]
I haven't tested this yet, but I doubt this works correctly when the Fiend attacks something while standing on tiberium, because `hidingplace && !moving` would still be fulfilled.

We might need #13736 with long enough `RevokeDelay` cooldown to cover that.

**Comment by pchote** [comment]
This could also be implemented using the same plumbing as the `TakeCover` trait.

Did crouching do anything else, like make it unselectable/disable tooltips/give armor bonuses?

**Comment by reaperrr** [comment]
> Did crouching do anything else, like make it unselectable/disable tooltips/give armor bonuses?

Not as far as I know, except that maybe it only healed on tiberium while laying down (I'd need to test that on the original to be sure, though).

**Comment by Mailaender** [comment]
> This could also be implemented using the same plumbing as the `TakeCover` trait.

It doesn't support standup and liedown sequences. https://github.com/OpenRA/OpenRA/issues/6330

**Comment by pchote** [comment]
Neither does switching out the rendering traits.  If we do need those then `IRenderInfantrySequenceModifier` will eventually support them, but the current approach will probably never work.

**Comment by pchote** [comment]
Dependencies have been merged.  Needs a rebase.

**Comment by Mailaender** [comment]
Rebased.

**Comment by pchote** [comment]
The shards on the fiend's backs are neutral grey rather than tiberium green.  Is this correct?

screen shot 2017-12-28 at 16 20 29

**Comment by reaperrr** [comment]
> Is this correct?

They were always tiberium-colored in the original, so no.

**Comment by pchote** [comment]
I suppose they use the same palette remapping as the tiberium does.  Setting them up with the tiberium palette should be easy.

**Comment by Mailaender** [comment]
Fixed that as well.

**Review by reaperrr — DISMISSED** [review]

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
Note: I've added a small fix for the fiend's editor category (it was in Infantry instead of Critter).

**Comment by reaperrr** [comment]
@OpenRA/engine-hackers Could some take a look at this and #13934 so we can get them merged?
You can ignore the AppVeyor failure here, it just failed to download a dependency.

**Review by GraionDilach — APPROVED** [review]
:+1:

**Pull request merged**